pandas/tests/frame/test_dtypes.py

```diff
@@ -622,3 +623,75 @@ def test_astype_str(self):
                         'NaT                       NaT' in result)
         self.assertTrue('2 2013-01-03 2013-01-03 00:00:00-05:00 '
                         '2013-01-03 00:00:00+01:00' in result)
+
+    def test_values_is_ndarray_with_datetime64tz(self):
+        df = DataFrame({
+            'A': date_range('20130101', periods=3),
+            'B': date_range('20130101', periods=3, tz='US/Eastern'),
+        })
+
+        for col in [
+            ["A"],
+            ["A", "A"],
+            ["A", "B"],
+            ["B", "B"],
+            ["B"],
+        ]:
+            arr = df[col].values
+            fst_elem = arr[0, 0]
+
+            self.assertEqual(type(arr), np.ndarray)
+            if "B" in col:
+                self.assertEqual(type(fst_elem), pd.Timestamp)
+            else:
+                self.assertEqual(type(fst_elem), np.datetime64)
+
+    def test_values_dtypes_with_datetime64tz(self):
+        df = DataFrame({'dt': date_range('20130101', periods=3),
+                        'dttz': date_range('20130101', periods=3,
+                                           tz='US/Eastern'),
+                        'td': (date_range('20130102', periods=3) -
+                               date_range('20130101', periods=3)),
+                        'cat': pd.Categorical(['a', 'b', 'b']),
+                        'b': [True, False, False],
+                        'i': [1, 2, 3],
+                        'f': [1.3, 2, 3],
+                        'c': [1j, 2, 3],
+                        })
+
+        cols = itertools.chain(
+            itertools.combinations_with_replacement(df.columns, 1),
+            itertools.combinations_with_replacement(df.columns, 2)
+        )
+        for col in cols:
+
+            df_sub = df[list(col)]
```

Comment:
testing the logic of _interleaved_dtype by asking to test all cases of df.values on all combination of dtypes and verifying the expected dtype. I have adapted the test comments to make it more explicit